### ISSUE TITLE
Transformer and GNMT on English to French

### ISSUE THREAD

**Issue by bitfort** [mlcommons/training_policies#175]
There is a proposal to move from english-to-german to english-to-french. The dataset is larger, and we could get to a higher score. 

SWG Notes:
We agree this should be explored.

**Comment by bitfort** [comment]
SWG Notes:
See AIs in:
https://github.com/mlperf/policies/issues/171
https://github.com/mlperf/policies/issues/174

**Comment by David-Levinthal** [comment]
If you want higher scores..english to french is easier...so is german to
english
english to german is a "standard" test because it is hard
:-)
(thought there are certainly harder translations)
d

**Comment by keithachorn-intel** [comment]
**Background:** The team previously discussed a proposal to consider the much larger EN-FR dataset rather than the EN-DE dataset used for MLPerf v0.5 training submissions.  We discussed the impact of this dataset on measured accuracy and total training time (and cost) during that meeting.  The following compares the training characteristics and relative time-to-train of these two datasets.

**Data:** The following data was collected on AWS p3.16xlarge instances (8xV100) using the MLPerf v0.5 reference workload (PyTorch implementation).  The results of the EN-DE workload on this system were not quite as fast as the DGX-1 results submitted by NVIDIA, but close enough that they should be representative of optimized workload performance for the sake of this comparison.

Training Phase | EN-DE (AWS p3.16xlarge) |  EN-FR (AWS p3.16xlarge)
------------ | ------------- | -------------
Initial Setup | 34s | 229s
Test Cycle | 462s | 4382s
Validation Cycle | 2s | 2s
Test Cycle | 10s | 12s

As previously noted, the EN-FR dataset contains approximately 10x the number of samples as the EN-DE dataset, contributing to the ~10x longer training cycle.  The validation and test cycles are not considerably extended.  **However, the initial setup time (the time from initiation of the core workload script to the beginning of the first training phase was considerably extended by ~6.7x.**  The majority of time spent in the initial setup consists of pre-processing dataset data.  So with a much larger dataset, this pre-processing time has consequently extended.

It was the desire of EN-FR proponents that by extending the training cycle without substantially extending the evaluation cycles (ie, cycle overhead), that the performance of larger test configs would benefit.  However, under the current test methodology (total time to train) which captures initial overhead, combined with thresholds which are reached in only 3-4 epochs, the extended initial overhead considerably exceeds the cycle overhead reduction.

Comparing the p3.16xlarge results with NVIDIA's 32xDGX-1 results on the EN-DE dataset, it does **not** appear the initial setup overhead is reduced through parallelization.  So based on this I expect the initial overhead increase of EN-FR to similarly impact EN-FR, perhaps leading to worse measured performance, not better.

Training Phase | EN-DE (AWS p3.16xlarge) |  EN-DE (32xDGX-1, MLPerf v0.5 submission)
------------ | ------------- | -------------
Initial Setup | 34s | 35s
Test Cycle | 462s | 29s
Validation Cycle | 2s | (not logged)
Test Cycle | 10s | 6s

At this time, it appears switching to the EN-FR dataset would not only greatly extend the training time and cost incurred by submitters, but would also focus a greater percentage of the measured time on overhead rather than training.  If these findings hold, I don't believe this is the intended outcome of changing datasets.  If interest remains in the original proposal, more measurements on 32xDGX-1 or large TPU clusters may be conducted.

**Comment by bitfort** [comment]
SWG Notes:

Assuming resolution of the Initialization Overhead issue, we can agree to not move to english/french for either GNMT or Transformer for v0.6.

**Comment by bitfort** [comment]
No changes needed.
